packages/compiler/config/config-loader.ts

```diff
@@ -17,42 +18,53 @@ export const defaultConfig = deepFreeze({
 
 /**
  * Look for the project root by looking up until a `tspconfig.yaml` is found.
- * @param path Path to start looking
+ * @param path Path to the file or the folder to start looking
  */
 export async function findTypeSpecConfigPath(
   host: CompilerHost,
   path: string
 ): Promise<string | undefined> {
-  let current = path;
-  while (true) {
-    let pkgPath = await searchConfigFile(host, current, TypeSpecConfigFilename);
-    if (pkgPath === undefined) {
-      pkgPath = await searchConfigFile(host, current, OldTypeSpecConfigFilename);
-    }
-    // if found either file in current folder, return it
-    if (pkgPath !== undefined) {
-      return pkgPath;
-    }
-    const parent = getDirectoryPath(current);
-    if (parent === current) {
-      return undefined;
+  try {
+    const stats = fs.statSync(path);
+    if (stats.isFile()) {
+      return path;
+    } else if (stats.isDirectory()) {
+      let current = path;
+      while (true) {
+        let pkgPath = await searchConfigFile(host, current, TypeSpecConfigFilename);
+        if (pkgPath === undefined) {
+          pkgPath = await searchConfigFile(host, current, OldTypeSpecConfigFilename);
+        }
+        // if found either file in current folder, return it
+        if (pkgPath !== undefined) {
+          return pkgPath;
+        }
+        const parent = getDirectoryPath(current);
+        if (parent === current) {
+          return undefined;
+        }
+        current = parent;
+      }
     }
-    current = parent;
+    return undefined;
+  } catch (error) {
+    console.error(`ERROR: --config '${path}' not found. Using default configuration.`);
+    return undefined;
```

Comment:
@markcowl @nguerrera is this the right behavior?  Here's what it looks like in the IDE:

![image](https://user-images.githubusercontent.com/5723682/222844762-6a5b7430-2396-4fcd-a612-feb71c6f7ddc.png)

Reply:
It should be a diagnostic. Like when the entry point is not found:

```
$ tsp compile noexist.tsp
TypeSpec compiler v0.40.0

Diagnostics were reported during compilation:

error file-not-found: File C:/Src/typespec/noexist.tsp not found.

Found 1 error.
```

Reply:
In the existing behavior today it doesn't throw a diagnostic. It silently gives up and uses a default configuration. I figure it at least makes sense to show a warning in this case (though I see I made it an error). Making it fail might have unintended consequences, but leaving it as-is where it silently does something different also feels wonky.

Reply:
As long as we differentiate between a passed-in configuration path and the default configuration, I think it is OK to emit an error diagnostic.  We should not emit an error diagnostic because someone ran cadl without a configuration file.  We *should* emit an error diagnostic if someone passed in a configuration path and we could not open the specified file.

Reply:
Changed this to an error diagnostic.